### PR TITLE
feat(sdk): summarize jira posture graph layering

### PR DESCRIPTION
**Pull request by jonathanhaaswriter**
## Summary
- add graph summaries on top of the raw Jira posture neighborhood output in both SDK examples
- count unique node types and relation types, track per-root neighborhood sizes, and preserve any graph fetch errors in the summary
- make the posture examples return both the raw graph layering and a concise graph-summary view that is easier to inspect during onboarding

## Testing
- make verify
- python3 -m py_compile sdk/python/examples/jira_posture_onboarding.py
- PYTHONPATH=sdk/python python3 smoke test for summarize_graph_layering
- cd sdk/typescript && npx -y -p typescript tsc -p tsconfig.json
- cd sdk/typescript && npx -y -p tsx tsx smoke test for summarizeGraphLayering

### CODE REVIEW

**Comment by jonathanhaaswriter** [comment]
@droid Please perform a deep code review and security review of this PR. Leave inline comments only for high-confidence, actionable issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review after the latest pushed fixes. Focus on whether the prior findings on this PR are resolved and only report remaining high-confidence issues.

**Comment by jonathanhaaswriter** [comment]
### Droid sweep status

Latest sweep: `2026-04-29 22:09 UTC`.

- Unresolved non-outdated review threads: `0`
- CI `verify`: `SUCCESS`
- `droid-review`: `not present`

I’ll keep watching for new review findings/check failures and address them as they appear.

**Comment by jonathanhaaswriter** [comment]
Droid sweep audit complete (final pass)

- Re-scanned all live review threads, including outdated ones
- Verified each remaining thread against the current branch tip
- Resolved every actionable thread (or fixed it inline and pushed)
- 0 unresolved review threads across the whole open-PR stack at this time

PR #494 (fix/feedback-main-followup) and PR #495 (feat/droid-creation-workflows) received additional fixes for SSRF (DNS-rebind) and workflow branch-name validation in this pass.

**Comment by jonathanhaaswriter** [comment]
@droid review

**Comment by jonathanhaaswriter** [comment]
@droid please harden this PR end-to-end:

1. Re-review the diff for residual bugs, races, edge cases, security gaps, and untested paths.
2. Apply the smallest correct fixes for any high-confidence issues you find.
3. Strengthen test coverage: add regression tests for every prior review finding on this PR plus the new behavior, including failure paths, boundary inputs, concurrency, and not-found / typed-nil interface guards.
4. Run focused tests and `make verify`. If validators fail, fix until green.
5. Commit and push the fixes to this PR's branch (or, if you lack push permission in this workflow, leave concrete inline suggestions and a summary comment).

Do not change the PR scope, do not modify the default branch, and do not merge.

**Comment by jonathanhaaswriter** [comment]
@droid review

File: sdk/python/examples/jira_posture_onboarding.py [hunk 1]
```diff
@@ -262,6 +263,69 @@ def load_graph_layering(integration: IntegrationClient, posture: Dict[str, Any])
...
+        for node in [root] + list(entry.get("neighbors", [])):
+            if not isinstance(node, dict):
+                continue
+            node_urn = optional_string(node.get("urn"))
+            entity_type = optional_string(node.get("entity_type")) or "unknown"
+            if node_urn is None or node_urn in seen_nodes:
+                continue
+            seen_nodes.add(node_urn)
+            node_counts[entity_type] = node_counts.get(entity_type, 0) + 1
+        for relation in entry.get("relations", []):
```
[the remark this answers is not in the document]
Comment (jonathanhaaswriter) [reply sdk/python/examples/jira_posture_onboarding.py]: Fixed in a7ab6b612 by normalizing non-list/null `neighbors` and `relations` to empty lists before measuring or iterating them; also smoke-tested the example with null arrays.

**Comment by jonathanhaaswriter** [comment]
@droid review

Please re-review the Jira posture graph-layering summary deeply. Summaries must be deterministic and trustworthy or they become a footgun:

1. **Counting determinism**: verify unique node-type and relation-type counts are deterministic across runs and platforms (map iteration, locale, ordering), and that ties/repeats are handled identically in Python and TS.
2. **Per-root neighborhood sizing**: confirm per-root sizes are computed against the bounded neighborhood (PR 460) and never reflect unbounded enumeration; check what happens when a root returns no nodes or only the root itself.
3. **Error preservation**: ensure graph-fetch errors are preserved per-root in the summary in a structured way (typed error fields, not free-form strings) and that the summary still renders meaningfully when some roots fail.
4. **Output shape and stability**: verify the summary view exposes a stable, additive shape so downstream PRs (470 helpers, 471 findings) can rely on field names/types without breaking on future fields.
5. **Information disclosure**: re-check that summarized errors do not leak internal store paths, query text, or tenant data, and that node-type/relation-type strings come from a controlled vocabulary.
6. **Edge cases**: check empty graphs, single-node neighborhoods, duplicate roots, and very-large fan-out roots to make sure summary output is sensible (no NaN, no negative counts, no off-by-one in “unique” computations).
7. **Python/TS parity**: ensure both SDK examples produce equivalent summaries for the same neighborhood input, and that any divergence is caught by tests rather than discovered by integrators.

**Comment by jonathanhaaswriter** [comment]
Closing this PR — its substantive feature work is already in `main` via the earlier consolidation rollups, and the unresolved factory-droid findings posted on this PR have been addressed in main via the follow-up PRs:

- #497 (`fix(github): drop non-cursor opaque from terminal audit checkpoints`)
- #499 (`fix: address post-merge feedback across pending stacks`)

PR #499 specifically addresses the findings posted on this stack:

- Reject malformed `root_urn` values in `GetEntityNeighborhood` (PR #460)
- Map `refuted` and `superseded` claim statuses to projection deletion (PR #463)
- Tenant-scope `ListClaims` reads (PR #464)
- Per-runtime serialization of `WriteClaims` (PR #474)
- Restore null-array normalization in `graph_summary` (PR #470)
- Surface blank-root errors from `graphLayering` (PR #468)

This branch has diverged ~50k lines from `main` and rebasing it cleanly is no longer practical; closing here keeps the open queue accurate. If anything from this stack still needs to land, please open a fresh PR cut from `main`.
